src/drive/web/modules/views/Sharings/FilesViewerSharings.jsx

```diff
@@ -0,0 +1,50 @@
+import React from 'react'
+import { useQuery } from 'cozy-client'
+import { useRouter } from 'drive/lib/RouterContext'
+import { buildSharingsQuery } from 'drive/web/modules/queries'
+import { getCurrentFileId } from 'drive/web/modules/selectors'
+import { connect } from 'react-redux'
+import FilesViewer, {
+  FilesViewerLoading
+} from 'drive/web/modules/viewer/FilesViewerV2'
+import SharedDocuments from 'cozy-sharing/dist/components/SharedDocuments'
+
+const FilesViewerWithQuery = ({ sharedDocumentIds, ...props }) => {
+  const filesQuery = buildSharingsQuery(sharedDocumentIds)
+  const results = useQuery(filesQuery.definition, filesQuery.options)
+  const { router } = useRouter()
+  if (results.data) {
+    const viewableFiles = results.data.filter(f => f.type !== 'directory')
+    return (
+      <FilesViewer
+        {...props}
+        files={viewableFiles}
+        filesQuery={results}
+        onClose={() => router.push('/sharings')}
+        onChange={fileId =>
+          router.push({
+            pathname: '/sharings/file/' + fileId
+          })
+        }
+      />
+    )
+  } else {
+    return <FilesViewerLoading />
+  }
+}
+
+const mapStateToProps = state => ({
+  fileId: getCurrentFileId(state)
+})
+
+const FilesViewerWithFolderId = connect(mapStateToProps)(FilesViewerWithQuery)
+
+const SharingsContainer = props => (
+  <SharedDocuments>
+    {({ sharedDocuments }) => (
+      <FilesViewerWithFolderId {...props} sharedDocumentIds={sharedDocuments} />
+    )}
+  </SharedDocuments>
+)
+
+export default SharingsContainer
```

Comment:
Nit: I find it always a bit surprising when a file's default export is not named as the file's name.

Nit: It seems a withSharedDocuments HOC could be shared between this file and Sharings/index.jsx file